File: BaekJoon/Python/31247.py
T = int(input())

N = [0]*T
K = [0]*T

for i in range(T):
    N[i], K[i] = map(int,input().split())


def cal_division(n):
    data = []
    for i in range(1,n+1):
        if n % i == 0:
            data.append(i)
    return data    


for i in range(T):
    n = N[i]
    k = K[i]
    
    count_k = 0
    while n > 0:
        arr = cal_division(n)
        # print(arr)
        count_even = 0
        count_odd = 0
        for ele in arr:
            if ele % 2 == 0:
                count_even += 1
            else:
                count_odd += 1
        # print(f'even : {count_even}, odd : {count_odd}')
        if k*count_odd == count_even:
            # print(f'MATCH!!! {count_even} = {k}*{count_odd}')
            count_k += 1 
        n -= 1
    print(count_k)

File: BaekJoon/C++/31247.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<int> cal_division(int n)
{
    vector<int> result;
    for(int i = 1; i < n+1; i++)
    {
        if(n % i == 0)
        {
            result.push_back(i);
        }
    }

    return result;

}

int main()
{
    int T;
    cin >> T;

    vector<int> N(T);
    vector<int> K(T);

    for(int i = 0; i < T; i++)
    {
        cin >> N[i] >> K[i];
    }

    for(int i = 0; i < T; i++)
    {
        int n = N[i];
        int k = K[i];

        int count_k = 0;
        while(n > 0)
        {
            int count_even = 0;
            int count_odd = 0;
            vector<int> arr = cal_division(n);
            for(int j = 0; j < arr.size(); j++)
            {
                if(arr[j] % 2 == 0)
                    count_even++;
                else
                    count_odd++;
            }

            if(count_even == k*count_odd)
                count_k++;
            n--;
        }
        cout << count_k << endl;
    }

    return 0;
}